lib/publiccloud/instance.pm

```diff
@@ -264,6 +264,33 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+=head2 verify_upload_logs
+
+    verify_upload_logs();
+
+Check files status and upload logs ok to UI.
+=cut
+
+sub verify_upload_logs {
+    my ($self, $dir, @files) = @_;
```

Comment:
This function has the limitation the all the files must reside in the same directory...  I'd rather drop the `$dir` argument and let the caller do the `join(' ', map { "$dir/$_" } @logs)` dance when calling this function.

Reply:
I'd even agree, but yes this specific routine is for logs files and is atomic for one directory with many files.

Generally logs are placed in one specific directory or few other, but using the full_name as entry, we'd manage long path strings and our main test logs also will report long full-file-names, reducing visibility and increasing log pollution.

Instead, in case of eventual (few) different more directories, we can call that atomic routine for each one, in a modular loop structure.

Reply:
@m-dati We have 2 tickets for code cleanup.  One for removing dead code and another for refactoring.  It's for the containers library.

Please don't make the public cloud library more complicated than needed.  Let's keep the KISS principle, specially in libraries.

Reply:
Still, you can also set empty dir "" and use full path names.

Reply:
> Still, you can also set empty dir "" and use full path names.

Still an anti-pattern.  `$dir` can be removed without impacting anything else in this function.

Reply:
$dir removed from input

Reply:
> $dir removed from input

Forgot to force-push?

Reply:
$dir removed as routine input, but same name used in the calling module, to contain the directory, to sew it with logs names in an array, that we submit to that routine (like in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19261#discussion_r1595183665)